added imagepullsecret to the CRD/pod creation

prune998 opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes https://github.com/banzaicloud/kafka-operator/issues/106
| License         | Apache 2.0


### What's in this PR?
Added `ImagePullSecrets` parameter to the `KafkaCluster` CRD so we can define the secrets needed to pull the Container image from a private repository.


### Why?
see Issue https://github.com/banzaicloud/kafka-operator/issues/106

### Checklist
<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->

- [ ] Implementation tested
- [ ] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [ ] Logging code meets the guideline
- [ ] User guide and development docs updated (if needed)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/banzaicloud/kafka-operator?pullRequest=108) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/banzaicloud/kafka-operator?pullRequest=108) before we can accept your contribution.<br/><hr/>**prune** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/banzaicloud/kafka-operator?pullRequest=108) it.</sub>

prune998 commented:
I just added it.
As a side note, the `Deployments` used by Envoy and CC are really basics. Even the `ServiceAccount` wasn't set in them.
I added it for now, but one could decide to use it's own Kafka image while using public onces for CC or Envoy.
If we don't want to be too limitative, we should allow to use different SA and ImagePullSecrets for each one of them... making a more complex `CRD`...

Let me know what you think while reviewing this commit.

Thanks.

prune998 commented:
Maybe I should also update `config/crds/banzaicloud_v1alpha1_kafkacluster.yaml` and `charts/kafka-operator/templates/operator-kafka-crd.yaml` to reflect the change  ? 
Latest K8s versions may reject the CR if they contains fields not explecitely defined in the CRD... don't remember which...

(I'm new to Kubebuilder, and I'm just testing the V2 now, I was previously using coreOS Operator SDK, please don't blame me for my ignorance... :) )

prune998 commented:
BTW, I already signed the CLA, and I just did again...

baluchicken commented:
No worries, I am just testing the v2 version as well. Check the #105 

You are right please run the `make manifests` and `make generate` to regenerate the crd. It will regenerate `config/crds/banzaicloud_v1alpha1_kafkacluster.yaml` but not the one in dir `chart`. Please update that with the help of the first one.

We are in a middle of redesigning the whole CRD. It is indeed a good point to allow to use different `ServiceAccount` and `ImagePullSecrets`. I am thinking to use `PodSpec` in the CRD for Brokers, Envoy, CruiseControl. With that we can get every field which is configurable but the CRD does not get too messy.

baluchicken commented:
@prune998 yes all commits should be verified, but no worries I will merge this as is, but next time please use the right user. Thanks
